File: Day214_2025/Q214_Leetcode.py
from collections import Counter
from typing import List

class Solution:
    def minCost(self, basket1: List[int], basket2: List[int]) -> int:
        freq = Counter()
        m = float('inf')

        for b1 in basket1:
            freq[b1] += 1
            m = min(m, b1)

        for b2 in basket2:
            freq[b2] -= 1
            m = min(m, b2)

        merge = []
        for fruit, count in freq.items():
            if count % 2 != 0:
                return -1
            merge.extend([fruit] * (abs(count) // 2))

        merge.sort()
        half = len(merge) // 2
        return sum(min(2 * m, merge[i]) for i in range(half))

File: Day214_2025/Q214_Leetcode.java
import java.util.*;

class Q214_Leetcode {
    public long minCost(int[] basket1, int[] basket2) {
        int m = Integer.MAX_VALUE;
        Map<Integer, Integer> freqMap = new HashMap<>();

        for (int b1 : basket1) {
            freqMap.put(b1, freqMap.getOrDefault(b1, 0) + 1);
            m = Math.min(m, b1);
        }

        for (int b2 : basket2) {
            freqMap.put(b2, freqMap.getOrDefault(b2, 0) - 1);
            m = Math.min(m, b2);
        }

        List<Integer> merge = new ArrayList<>();
        for (Map.Entry<Integer, Integer> entry : freqMap.entrySet()) {
            int val = entry.getKey();
            int count = entry.getValue();

            if (count % 2 != 0)
                return -1;

            for (int i = 0; i < Math.abs(count) / 2; i++) {
                merge.add(val);
            }
        }

        Collections.sort(merge);
        long cost = 0;
        int half = merge.size() / 2;
        for (int i = 0; i < half; i++) {
            cost += Math.min(2L * m, merge.get(i));
        }

        return cost;
    }
}
